rfc/0020-table-value-types/README.md

```diff
@@ -0,0 +1,197 @@
+<!--
+SPDX-FileCopyrightText: 2025 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0020: Value types as table schema <!-- TODO: adapt title -->
+
+| | |
+|---|---|
+| Feature Tag | `table-schema-value-types` |
+| Status | `DRAFT` |
+| Responsible | `jrentlez` |
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+This RFC changes the way sheets are parsed into tables, using a multi property
+value type as the table schema definition.
+
+## Motivation
+
+- Allows users to define how a sheet is parsed into a table.
+- Allows users to reuse table schema definitions.
+- Remove special syntax regarding table definitions.
+
+## Explanation
+
+From now on, value types are used to define a table schema, each property
+defining a column.
+```jayvee
+valuetype Coordinate {
+  x oftype integer;
+  y oftype integer;
+}
+
+block CoordinateTable oftype TableInterpreter {
+  columns: Coordinate;
+}
+```
+The table output by `CoordinateTable` has two columns `x` and `y`, both
+containing integer values.
+
+However, this creates problems if the sheet contains column names with spaces,
+because such a column could not be represented as a value type property.
+Thus, parsing the individual rows must be done explicitly by the user, inside a
+transform (`<>` denotes a placeholder):
+
+```jayvee
+transform <transform> {
+  from <row> oftype SheetRow;
+  to <output> oftype <value type>;
+
+  <output>: <row-creation expression>
+}
+
+block <block> oftype TableInterpreter {
+  columns: <value type>;
+  rowParsedWith: <transform>;
+}
+```
+
+Because `<transform>` parses rows, it needs to conform to:
+- only one input of type `SheetRow`.
+- only one output with the same value type as in `<block>`'s `columns` property.
+- must use a row-creation expression (see below)
+
+### `SheetRow`
```

Comment:
Foreshadow a little bit at the beginning that we need to introduce multiple concepts (otherwise this seems to be random as if it should belong into another RFC). Have a list of things this RFC introduces at the start:
- A new syntax to define columns of a table via a valuetype
- SheetRow builtin value type with new cell/field access syntax
- - ...